### PR TITLE
V22F-875 Internal meta data

### PR DESCRIPTION
**Pull request by noutheesink**
Opens meta data buffer file after saving, if buffer exists. Saves trial ID in buffer file. Shuts down black box after saving to database

### CODE REVIEW

**Review by larsdekwant — CHANGES_REQUESTED** [review]

**Review by KungFuDonkey — APPROVED** [review]
Can't find anything next to what @larsdekwant has said, so if that is changed I would approve, so I approve now

**Review by KungFuDonkey — APPROVED** [review]

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/SQLDatabaseStorage.cpp [hunk 1]
```diff
@@ -5,6 +5,8 @@
 #include "ConfigEnums.h"
 #include <config.h>
 
+#define META_BUFFER_FILENAME "sda_blackbox_internal_metadata_buffer.bin"
```
Comment (larsdekwant) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/SQLDatabaseStorage.cpp:8]: Move this to `ConfigEnums.h`, this is where I defined all the other buffer files as well, this also allows usage in multiple files.

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backendUnitTests/SQLDatabaseStorageTests.cpp [hunk 2]
```diff
@@ -172,6 +172,60 @@ TEST(SQLDatabaseStorageTests, TestRemoteCorrectFakeCert)
     ASSERT_NO_THROW(sqlDatabaseStorage.Run("test_file.txt", TEST_DATA_DIRECTORY "remote" OS_SEPARATOR "correctRemote"));
 }
 
+/// @brief Tests if the first 4 bytes in buffer file are correctly overwritten by SaveTrialIdToMetadata
+TEST(SQLDatabaseStorageTests, SaveTrialIdTest)
+{
+    // delete buffer file to make sure it doesn't exist
+    filesystem::path bufferPath = filesystem::temp_directory_path();
+    bufferPath.append("sda_blackbox_internal_metadata_buffer.bin");
```
Comment (larsdekwant) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backendUnitTests/SQLDatabaseStorageTests.cpp:180]: Replace this once you moved it to `ConfigEnums.h`. This makes sure that there can never be a miss-match between these two hardcoded filenames.
```suggestion
    bufferPath.append(META_BUFFER_FILENAME);
```

**Review by larsdekwant — CHANGES_REQUESTED** [review]

**Review by larsdekwant — COMMENTED** [review]
Resolve merge conflicts first

**Review by larsdekwant — APPROVED** [review]
Seems to work correctly, only issue is that this will always get saved no matter what, but I guess that's fine since it was a last moment request from the client...

**Pull request merged**